packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -196,6 +201,47 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     }
   }
 
+  /**
+   * Finds the next valid Dropdown option and sets the selected index to it.
+   * @param stepValue Value of how many items the function should traverse.  Should be -1 or 1.
+   * @param index Index of where the search should start
+   * @param selectedIndex The selectedIndex Dropdown's state
+   * @returns The next valid dropdown option's index
+   */
+  private _moveIndex(stepValue: number, index: number, selectedIndex: number): number {
+    // Return selectedIndex if nothing has changed
+    if (selectedIndex === index) {
+      return selectedIndex;
+    }
+    const { options } = this.props;
+    let stepCounter = 0;
+
+    // Set starting index to 0 if index is < 0
+    if (index < 0) {
+      index = 0;
+    }
+    // Set starting index to last option index if greater than options.length
+    if (index >= options.length) {
+      index = options.length - 1;
```

Comment:
can there be a case when options.length is 0? Then we'd set the index to -1 and then try to index in line 228